### ISSUE TITLE
Configuring permissions on file sink files

### ISSUE THREAD

**Issue by ptrdlbrg** [vectordotdev/vector#10727]
I'm running Vector in a Docker container on Linux. I'm using the vector file sink and I was looking to configure what permissions the output files have.  The default file permissions allow all users to read the log files and I'd like to remove that permission for some log files.

The [file sink documentation](https://vector.dev/docs/reference/configuration/sinks/file/) has no file mode configuration options so I assume that's not currently possible via the configuration. 

I can set the umask for Vector process but that has the downside of being global rather than allowing different permissions for different file sinks. To make that work under Docker I have to build a new Docker image with a short script that sets the umask before running vector. Its a bit inconvenient, but doable.

Can the file permissions be made configurable per file sink?

Failing that, can a global option be added to Vector to allow setting the umask? It'd be nicer to set it in the Vector configuration instead of setting it before running Vector. Since log files may contain sensitive data, it may be a good idea to default the umask to 027 to prevent the files from being world-readable by default.

For the benefit of others in my situation, it would be good to update the file sink documentation to mention:
1. The default file permissions are world readable
2. The default file permissions can be modified globally by setting umask before running vector
3. Another workaround is to set permissions on the parent directory to restrict access (e.g. chmod parent directory permissions to 750 instead of 755). This isn't identical to setting the permissions on files, but may be good enough for some use-cases.

**Comment by spencergilbert** [comment]
I don't believe this is configurable today, but seems like a very reasonable improvement to make 👍

**Comment by mikezsin** [comment]
@spencergilbert any plans to make this very reasonable improvement)?
